sdk/dapp-kit/src/utils/walletUtils.ts

```diff
@@ -2,18 +2,19 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import type {
-	MinimallyRequiredFeatures,
 	Wallet,
 	WalletWithFeatures,
+	MinimallyRequiredFeatures,
 } from '@mysten/wallet-standard';
-import { isWalletWithRequiredFeatureSet } from '@mysten/wallet-standard';
-import type { StorageAdapter } from './storageAdapters.js';
+import { getWallets, isWalletWithRequiredFeatureSet } from '@mysten/wallet-standard';
 
-export function sortWallets<AdditionalFeatures extends Wallet['features']>(
-	wallets: readonly Wallet[],
+export function getRegisteredWallets<AdditionalFeatures extends Wallet['features']>(
 	preferredWallets: string[],
 	requiredFeatures?: (keyof AdditionalFeatures)[],
 ) {
+	const walletsApi = getWallets();
```

Comment:
Is there any downside for having this re-initialized multiple times? I'd maybe dive into the implementation and just verify that there's not (I think we already call it multiple times today as well, I just want to verify before we double down).

Reply:
Nope, if `wallets` has already been initialized  then it's just returns the frozen object

<img width="901" alt="image" src="https://github.com/MystenLabs/sui/assets/7453188/d8af4ff0-a3cd-4edb-b0ed-73977e3baf97">